src/class/StoreContext.class.ts

```diff
@@ -0,0 +1,173 @@
+// Import Internal Dependencies
+import { TimedKVPeer, TimedKVPeerOptions } from "./TimedKVPeer.class";
+import { getRedis } from "../index";
+
+// CONSTANTS
+const kDefaultCookiesOptions: CookieSerializeOptions = { sameSite: "none", secure: true };
+const kStoreContextSessionName = "session-id";
+
+export interface CookieSerializeOptions {
+  domain?: string;
+  encode? (val: string): string;
+  expires?: Date;
+  httpOnly?: boolean;
+  maxAge?: number;
+  path?: string;
+  sameSite?: boolean | "lax" | "strict" | "none";
+  secure?: boolean;
+  signed?: boolean;
+}
+
+export interface UseContext<T> {
+  initSession: (id: string, payload: T) => Promise<string>;
+  destroySession: () => Promise<void>;
+  getSession: () => Promise<T | null>;
+  updateSession: (payload: Partial<T>) => Promise<void>;
+  isUserAuthenticated: () => Promise<boolean>;
+}
+
+export interface Store {
+  /** Return callback URL as string **/
+  returnTo?: string;
+}
+
+export interface FrameworkContext {
+  getCookie: (cookieName: string) => string;
+  setCookie: (cookieName: string, cookieValue: string | null, opts?: CookieSerializeOptions) => void
+}
+
+export interface StoreContextOptions<T extends Store> extends TimedKVPeerOptions<T> {
+  /** Property name used in isUserAuthenticated() method to define if the user is authenticated or not **/
+  authentificationField: keyof T;
+  /** HTTP Cookies options. Will be used when creating the session cookie. **/
+  setCookiesOptions?: CookieSerializeOptions;
+}
+
+/**
+* @class StoreContext
+* @description A session store designed to preserve a session between several Node.js instances
+*
+* @property {keyof T | null} authenticationField - if a specific field is specified for authentication, it will be provided here.
+* @property {CookieSerializeOptions} cookiesOptions - the options for the cookies
+*
+* @example
+* export interface DextStore extends Store {
+*   mail?: string;
+*   password?: string;
+* }
+*
+* export const store = new StoreContext<DextStore>({
+*   authentificationField: "mail"
+* });
+*/
```

Comment:
```suggestion
* @example
* ```ts
* export interface DextStore extends Store {
*   mail?: string;
*   password?: string;
* }
*
* export const store = new StoreContext<DextStore>({
*   authentificationField: "mail"
* });
* ```
*/
```
Si tu veux faire les choses vraiment bien il faut faire comme ça, essais tu verras la diff dans ton IDE ^^